client_test.go

```diff
@@ -2840,3 +2842,118 @@ func TestHttpsRequestWithoutParsedURL(t *testing.T) {
 		t.Fatalf("https requests with IsTLS client must succeed")
 	}
 }
+
+// See issue #1232
+func TestRstConnResponseWhileSending(t *testing.T) {
+	const expectedStatus = http.StatusTeapot
+	const payload = "payload"
+
+	srv, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	go func() {
+		conn, err := srv.Accept()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Read at least one byte of the header
+		// Otherwise we would have an unsolicited response
+		_, err = ioutil.ReadAll(io.LimitReader(conn, 1))
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Respond
+		_, err = conn.Write([]byte("HTTP/1.1 418 Teapot\r\n\r\n"))
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Forcefully close connection
+		err = conn.(*net.TCPConn).SetLinger(0)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		conn.Close()
+	}()
+
+	svrUrl := "http://" + srv.Addr().String()
+
+	client := HostClient{Addr: srv.Addr().String()}
+
+	req := AcquireRequest()
+	defer ReleaseRequest(req)
+	resp := AcquireResponse()
+	defer ReleaseResponse(resp)
+
+	req.Header.SetMethod("POST")
+	req.SetBodyStream(strings.NewReader(payload), len(payload))
+	req.SetRequestURI(svrUrl)
+
+	err = client.Do(req, resp)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	if expectedStatus != resp.StatusCode() {
+		t.Fatalf("Expected %d status code, but got %d", expectedStatus, resp.StatusCode())
+	}
+}
+
+// See issue #1232
+func TestRstConnClosedWithoutResponse(t *testing.T) {
+	const payload = "payload"
+
+	srv, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	go func() {
+		conn, err := srv.Accept()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Read at least one byte of the header
+		// Otherwise we would have an unsolicited response
+		_, err = ioutil.ReadAll(io.LimitReader(conn, 1))
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Respond with incomplete header
+		_, err = conn.Write([]byte("Http"))
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		// Forcefully close connection
+		err = conn.(*net.TCPConn).SetLinger(0)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		conn.Close()
+	}()
+
+	svrUrl := "http://" + srv.Addr().String()
+
+	client := HostClient{Addr: srv.Addr().String()}
+
+	req := AcquireRequest()
+	defer ReleaseRequest(req)
+	resp := AcquireResponse()
+	defer ReleaseResponse(resp)
+
+	req.Header.SetMethod("POST")
+	req.SetBodyStream(strings.NewReader(payload), len(payload))
+	req.SetRequestURI(svrUrl)
+
+	err = client.Do(req, resp)
+
+	if !isConnectionReset(err) {
+		t.Fatalf("Expected connection reset error")
```

Comment:
```suggestion
		t.Fatal("Expected connection reset error")
```

Reply:
Thanks! Did change all formatted strings to not formatted in tests as suggested.